src/main/java/dinkplugin/notifiers/data/DiaryNotificationData.java

```diff
@@ -1,13 +1,68 @@
 package dinkplugin.notifiers.data;
 
 import dinkplugin.domain.AchievementDiary;
+import dinkplugin.message.Field;
 import lombok.EqualsAndHashCode;
 import lombok.Value;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Value
 @EqualsAndHashCode(callSuper = false)
 public class DiaryNotificationData extends NotificationData {
+
+    /**
+     * The area of the diary that was just completed.
+     */
     String area;
+
+    /**
+     * The difficulty of the diary that was just completed.
+     */
     AchievementDiary.Difficulty difficulty;
+
+    /**
+     * The number of diaries that have been completed (across all regions and difficulties).
+     */
     int total;
+
+    /**
+     * The number of diary tasks that have been completed (across all regions and difficulties).
+     */
+    int tasksCompleted;
+
+    /**
+     * The total number of diary tasks within the game (across all regions and difficulties).
+     */
+    int tasksTotal;
+
+    /**
+     * The number of diary tasks completed within this specific area.
+     */
+    int areaTasksCompleted;
+
+    /**
+     * The total number of diary tasks possible within this specific area.
+     */
+    int areaTasksTotal;
+
+    @Override
+    public List<Field> getFields() {
+        List<Field> fields = new ArrayList<>(2);
+
+        if (areaTasksCompleted > 0 && areaTasksTotal > 0) {
+            fields.add(
+                new Field("Area Task Progress", Field.formatProgress(areaTasksCompleted, areaTasksTotal))
+            );
+        }
+
+        if (tasksCompleted > 0 && tasksTotal > 0) {
+            fields.add(
+                new Field("Overall Task Progress", Field.formatProgress(tasksCompleted, tasksTotal))
```

Comment:
Should `Overall Task Progress` be rephrased?

Reply:
I'm fine with this :+1:

Reply:
overall is fine but we'll probably make it `Overall Progress` to go with the other suggestion